landing-strip/core/src/main/java/com/novoda/landingstrip/TabsPagerListener.java

```diff
@@ -0,0 +1,85 @@
+package com.novoda.landingstrip;
+
+import android.support.v4.view.PagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.view.ViewGroup;
+
+import com.novoda.landingstrip.LandingStrip.TabSetterUpper;
+
+public class TabsPagerListener implements OnPagerAdapterChangedListener {
+    private final State state;
+    private final TabCreator tabCreator;
+    private final TabsContainer tabsContainer;
+
+    private PagerAdapterObserver pagerAdapterObserver;
+
+    public static TabsPagerListener newInstance(TabsContainer tabsContainer, Attributes attributes, ViewGroup parent, Scrollable scrollable) {
+        State state = State.newInstance();
+
+        state.updateDelegateOnPageListener(new ViewPager.SimpleOnPageChangeListener());
+        state.updatePosition(0);
+        state.updatePositionOffset(0);
+        state.invalidateFastForwardPosition();
+
+        TabCreator tabCreator = new TabCreator(attributes, state, tabsContainer, parent, scrollable);
+
+        tabsContainer.attachTo(parent);
+
+        TabsPagerListener tabsPagerListener = new TabsPagerListener(state, tabsContainer, tabCreator);
+        PagerAdapterObserver observer = new PagerAdapterObserver(tabsPagerListener);
+        tabsPagerListener.setPagerAdapterObserver(observer);
+        return tabsPagerListener;
+    }
+
+    public Coordinates calculateIndicatorCoordinates() {
```

Comment:
This _listener_ has a method called `calculate..()` which is delegated to a _state_ object, which internally has a calculator.